runtime/oti/j9nonbuilder.h

```diff
@@ -3042,29 +3042,84 @@ typedef struct J9IndexableObjectContiguous {
 #if defined(J9VM_ENV_DATA64) && defined(OMR_GC_FULL_POINTERS)
 	U_32 padding;
 #endif /* J9VM_ENV_DATA64 && !OMR_GC_COMPRESSED_POINTERS */
+} J9IndexableObjectContiguous;
+
+typedef struct J9IndexableObjectContiguousCompressed {
+	U_32 clazz;
```

Comment:
Personally, now that you touching this, I'd convert the U_32 and UDATA types to uint32_t, uintptr_t etc, both for indexable and non-indexable header shapes (starting just a few lines above), and any code/macros that use them.

At least in OMR code, the later is more prevalent, but in J9 it's still a very mixed bag.

Let's hear if @tajila agrees?

Reply:
for J9 code we are still using the U_32 and UDATA variants

Reply:
OK, let's leave it then. In J9/GC code however we gradually convert them. There is lots of code/APIs  blending between OMR/GC and J9/GC  (through passing arguments and return values) and we try to unify that aspect.